change/@ni-nimble-components-df1cc26c-9976-477a-8fa0-f9b72492eaa0.json

```diff
@@ -0,0 +1,7 @@
+{
```

Comment:
Couple issues I noticed while playing with keyboard nav in the [latest build of the Angular app](https://60e89457a987cf003efc0a5b-rgzslqaxvj.chromatic.com/example-client-app/#/customapp)
1. tab to table so that focus is on the Select All checkbox. Then press UpArrow and focus seems to end up at an unknown location (pressing DownArrow / Tab / Shift-Tab doesn't bring focus back to the table). I didn't double check our spec-ed behavior but I was expecting we'd just keep focus on that item in the header, similar to how focus stays on the last row when you press DownArrow. And probably it could scroll the page up, which is what happens if the focus is on anything in the header other than Select All. 
   - all the above also applies to focusing the select checkbox for the last row and pressing DownArrow too
3. The behavior was unexpected when focusing something in the table, then tabbing to the Add Rows button, then Shift-Tabbing back to the table. If adding rows caused the set of visible rows to change, focus returns to a different data row than it was on before (albeit one in the same visible location). Screen recording below. The HLD says we'd return to the row with the same data index; I didn't actually check if the add rows button is shuffling around existing indices so maybe it's actually behaving as expected, but it felt weird to me. I don't think this is a showstopper if it's a challenge to get "ideal" behavior.

Reply:
https://github.com/ni/nimble/assets/10500124/d30a0819-f7e8-4bc7-8751-48315a8bea8b

Reply:
(2) is expected. The HLD says "based on row index" (not data ID) which is what happens - the 7th row in the table is focused when you Tab away from it, and when you tab back.

Reply:
(1) - I can't reproduce an issue here, a screen recording might help.
I'll note though, that if something in the header or last row is focused, and you keep pressing ArrowUp or ArrowDown more, the same item should stay focused, but the page will scroll too. If we wanted to we could prevent that by doing `preventDefault()` on those key events even if we're at the table extents already.

Reply:
Actually for (1) it's fine in Chrome, but I see what you described in Firefox. I'll take a look.

Reply:
1. I am indeed seeing this in Firefox.
2. I misinterpreted "row index" in the HLD to mean "data index". Sounds like the current behavior is to spec. 👍

Reply:
1 - fixed in current update. Also we'll no longer page-scroll on arrow up/down presses at table extents, as discussed.